src/Tronkell/Server/Server.hs

```diff
@@ -49,9 +50,9 @@ mainLoop server@Server{..} = do
 runClient :: Handle -> Server -> IO ()
 runClient clientHdl server@Server{..} = do
   hPutStr clientHdl "Take a nick name : "
-  nick <- cleanString <$> hGetLine clientHdl
-  let uId  = nickToUserId nick
-      user = User uId (Just . T.pack $ nick) Waiting
+  uId  <- modifyMVar serverLastUserId getNextUserID
+  nick <- Just . T.pack . cleanString <$> hGetLine clientHdl
```

Comment:
since you are getting `uId` and `nick` before creating a `User`.. it seems you do not need `Maybe` for `nick` .. :)